lab/torch/random.py

```diff
@@ -22,17 +22,25 @@ def global_random_state(_: TorchDType):
     else:
         parts = B.ActiveDevice.active_name.lower().split(":")
 
-        if len(parts) == 0 or parts[0] not in {"cuda", "gpu"}:
+        if len(parts) == 0 or parts[0] not in {"cuda", "gpu", "mps"}:
             raise RuntimeError(f'Unknown active device "{B.ActiveDevice.active_name}".')
 
-        # Ensure that the generators are available.
-        if len(torch.cuda.default_generators) == 0:
-            torch.cuda.init()
+        if parts[0] == "mps":
```

Comment:
```suggestion
        if parts[0] == "mps":
            if len(parts) != 1:
                raise ValueError("Cannot specify a device number for PyTorch MPS.")
```

Reply:
I think here even though you can only have one ups device torch still uses `"mps:0"` as the device name, so I changed this to `if parts[1] != "0"`

Reply:
Makes total sense :) Thanks!